test/gtest/uct/test_pending.cc

```diff
@@ -397,6 +396,7 @@ UCS_TEST_SKIP_COND_P(test_uct_pending, send_ooo_with_pending_another_ep,
             if (status != UCS_OK) {
                 ASSERT_EQ(UCS_ERR_NO_RESOURCE, status);
                 ep_pending_idx[idx] = true;
+                num_ep_pending++;
```

Comment:
what is the difference between n_pending and num_ep_pending?

Reply:
`num_ep_pending` - how many EPs were added to pending
`n_pending` - how many pending requests were added to pending

it could be not equal due to async progress, i.e. `n_pending` can be less then `num_ep_pending` if some pending request were completed before this loop is ended

Reply:
so "num_ep_pending" is actually "num_pending_added"?
maybe the skip condition could check n_pending?

Reply:
> so "num_ep_pending" is actually "num_pending_added"?

yes

> maybe the skip condition could check n_pending?

we are checking `(num_ep_pending < num_eps)`, if async progress completed some UCT pending request, then `n_pending < num_eps` will be always true

Reply:
can we check n_pending  > 0?

Reply:
we need to try fill resources for all EPs.
if we check for `n_pending > 0`, it closes the loop after one EP scheduled pending req

Reply:
but he check in line 418 is after the loop..

Reply:
@yosefe done as discussed